lib/jekyll.rb

```diff
@@ -49,12 +49,19 @@ module Jekyll
   autoload :Filters,             'jekyll/filters'
   autoload :FrontmatterDefaults, 'jekyll/frontmatter_defaults'
   autoload :Layout,              'jekyll/layout'
-  autoload :LayoutReader,        'jekyll/layout_reader'
+  autoload :CollectionReader,    'jekyll/readers/dynamic/collection_reader'
+  autoload :DataReader,          'jekyll/readers/dynamic/data_reader'
+  autoload :LayoutReader,        'jekyll/readers/dynamic/layout_reader'
+  autoload :DraftReader,         'jekyll/readers/dynamic/draft_reader'
+  autoload :PostReader,          'jekyll/readers/dynamic/post_reader'
+  autoload :PageReader,          'jekyll/readers/dynamic/page_reader'
+  autoload :StaticFileReader,    'jekyll/readers/static/static_file_reader'
```

Comment:
Why the distinction between dynamic vs. static? Why add that level of indirection? If we had more static readers, I wouldn't be asking about this, but it feels like YAGNI for just the one.